pkg/tenant/tenants_test.go

```diff
@@ -14,19 +16,78 @@ func (suite *tenantTestSuite) TestTenantCreate() {
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
+	// Connect to a mock trtl database
 	trtl := db.GetMock()
 	defer trtl.Reset()
 
+	// Call the OnPut method and return a PutReply
 	trtl.OnPut = func(ctx context.Context, pr *pb.PutRequest) (*pb.PutReply, error) {
 		return &pb.PutReply{}, nil
 	}
 
+	// Create a tenant test fixture
 	req := &api.Tenant{
 		Name:            "tenant01",
 		EnvironmentType: "prod",
 	}
+
 	tenant, err := suite.client.TenantCreate(ctx, req)
 	require.NoError(err, "could not add tenant")
 	require.Equal(req.Name, tenant.Name, "tenant name should match")
 	require.Equal(req.EnvironmentType, tenant.EnvironmentType, "tenant id should match")
 }
+
+func (suite *tenantTestSuite) TestTenantDetail() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	tenantID := "01ARZ3NDEKTSV4RRFFQ69G5FAV"
+	defer cancel()
+
+	// Connect to a mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Get JSON test data.
+	data, err := os.ReadFile("testdata/tenant.json")
+	if err != nil {
+		return
+	}
+
+	// Call the OnGet method and return the JSON test data.
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return &pb.GetReply{
+			Value: data,
+		}, nil
+	}
+
+	// Create a tenant test fixture.
+	req := &api.Tenant{
+		ID:              "01ARZ3NDEKTSV4RRFFQ69G5FAV",
+		Name:            "example-staging",
+		EnvironmentType: "prod",
+	}
+
+	tenant, err := suite.client.TenantDetail(ctx, tenantID)
+	require.Error(err, http.StatusBadRequest, "could not get tenant")
+	require.Equal(req, tenant, "tenant should match")
```

Comment:
Can we add a few quick cases here and in `TestTenantDelete` to cover the error cases? We just need to be sure that if the required params are not supplied we're getting the correct HTTP status response back with an error. We should also test that 404s are returned when the trtl mock is set up to return an error.